components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -852,6 +855,7 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
             log.debug("Retrieving information of access token(hashed): " + DigestUtils.sha256Hex
                     (accessTokenIdentifier));
         }
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
```

Comment:
Isn't there any other way we can get the tenant domain other than from the Thread Local? The concern is, if there is another tenant flow triggered, the tenant domain of the new tenant flow would be returned here instead of the intended one.